### ISSUE TITLE
No support for sources

### ISSUE THREAD

**Issue by ocharles** [dmjio/stripe#93]
It seems that the whole [sources API](https://stripe.com/docs/api#sources) is unimplemented. Is that correct?

**Comment by dmjio** [comment]
I don't believe so. At least, it wasn't possible to perform CRUD on sources in the 2014 version of the API. Although, there was some notion of a "source", but I think it referred to something different (like a parent object). This new source API seems to be related to payment methods.

Here is a list of endpoints this library is based on,
https://web.archive.org/web/20140802220818/https://stripe.com/docs/api#customer_object

We could try to construct a request (using the [stripe](https://hackage.haskell.org/package/stripe-haskell-2.3.0/docs/Web-Stripe-Client-Stripe.html) function) to the sources endpoint and see what happens.

**Comment by cutsea110** [comment]
Will you support payment APIs soon? 
That is, Payment APIs are PaymentMethod, PaymentIntents, SetupIntents.

**Comment by dmjio** [comment]
@cutsea110 Potentially, when do you need it by? https://github.com/dmjio/stripe/pull/122 has preliminary support for `PaymentIntents` as well. cc @NorfairKing

**Comment by NorfairKing** [comment]
@cutsea110 I made some code that's just enough for my purposes, but not really fit to be included into this library as-is.

**Comment by cutsea110** [comment]
@dmjio @NorfairKing Thanks for your quick reply.

I think I have to deal with deadline by next March.
I want to make the transition early as I have time to spare.

I found this ticket was opened at 2018, so I asked how this project will go.

Thanks!
